Jenkinsfile

```diff
@@ -392,6 +396,81 @@ pipeline {
                 }
             }
         }
+        stage('OpenCL tests'){
+            when {
+                expression {
+                    !skipRemainingStages
+                }
+            }
+            parallel {
+                stage('OpenCL CPU tests') {
+                    when {
+                        expression {
+                            !skipOpenCL
+                        }
+                    }
+                    agent { label "gelman-group-win2 || linux-gpu" }
+                    steps {
+                        script {
+                            if (isUnix()) {
+                                deleteDir()
+                                unstash 'StanSetup'
+                                sh "echo CXX=${env.CXX} -Werror > make/local"
+                                sh "echo STAN_OPENCL=true>> make/local"
+                                sh "echo OPENCL_PLATFORM_ID=${env.OPENCL_PLATFORM_ID_CPU}>> make/local"
+                                sh "echo OPENCL_DEVICE_ID=${env.OPENCL_DEVICE_ID_CPU}>> make/local"
+                                // skips tests that require specific support in OpenCL
+                                sh 'echo "ifdef NO_CPU_OPENCL_INT64_BASE_ATOMIC" >> make/local'
+                                sh 'echo "CXXFLAGS += -DSTAN_TEST_SKIP_REQUIRING_OPENCL_INT64_BASE_ATOMIC" >> make/local'
+                                sh 'echo "endif" >> make/local'
+                                runTests("lib/stan_math/test/unit/math/opencl", false)
```

Comment:
The test we will need to run are just

```cpp
runTests("src/test/unit")
```
We actually only need to run assign_cl and rvalue_cl, but these take just a few minutes so its fine.